src/QMCWaveFunctions/LCAO/LCAOrbitalSet.cpp

```diff
@@ -586,10 +701,55 @@ void LCAOrbitalSet::mw_evaluateDetRatios(const RefVectorWithLeader<SPOSet>& spo_
 
   mw_evaluateValueVPsImplGEMM(spo_list, vp_list, vp_phi_v);
 
+#if defined(ENABLE_OFFLOAD)
+  const size_t nw             = vp_list.size();
+  auto& invRow_deviceptr_list = spo_leader.mw_mem_handle_.getResource().invRow_deviceptr_list;
+  auto& ratios_buffer         = spo_leader.mw_mem_handle_.getResource().ratios_buffer;
+
+  invRow_deviceptr_list.resize(nVPs);
+  ratios_buffer.resize(nVPs);
+
+  std::vector<size_t> nVP_mw(nw, 0);  // number of VPs per walker
+  std::vector<size_t> iVP0_mw(nw, 0); // first VP idx of each walker
```

Comment:
Can we simply avoid `nVP_mw` and `iVP0_mw`? Feel the code potentially be simplified by modifying the loop body at line 721

Reply:
changed.